game-core/src/main/java/games/strategy/engine/message/UnifiedInvocationHandler.java

```diff
@@ -51,32 +51,25 @@ public Object invoke(final Object proxy, final Method method, final Object[] arg
 
     final RemoteMethodCallResults response = messenger.invokeAndWait(endPointName, remoteMethodMsg);
     if (response.getException() != null) {
-      if (response.getException() instanceof MessengerException) {
-        final MessengerException cle = (MessengerException) response.getException();
-        cle.fillInInvokerStackTrace();
-      } else {
-        // do not chain the exception, we want to keep whatever the original exception's class was, so just add our
-        // bit to the stack
-        // trace.
-        final Throwable throwable = response.getException();
-        final StackTraceElement[] exceptionTrace = throwable.getStackTrace();
-        final Exception ourException =
-            new Exception(throwable.getMessage() + " exception in response from other system");
-        // Thread.currentThread().getStackTrace();
-        final StackTraceElement[] ourTrace = ourException.getStackTrace();
-        if (exceptionTrace != null && ourTrace != null) {
-          final StackTraceElement[] combinedTrace = new StackTraceElement[(exceptionTrace.length + ourTrace.length)];
-          int i = 0;
-          for (final StackTraceElement element : exceptionTrace) {
-            combinedTrace[i] = element;
-            i++;
-          }
-          for (final StackTraceElement element : ourTrace) {
-            combinedTrace[i] = element;
-            i++;
-          }
-          throwable.setStackTrace(combinedTrace);
+      // do not chain the exception, we want to keep whatever the original exception's class was, so just add our
```

Comment:
I'm not sure if we actually want this kind of special treatment.
I'd be ok with completely removing it, I don't like that we essentially mess around with the stacktrace.

Reply:
Digging a little bit deeper I'm pretty sure we should just wrap the exception in a runtimeexception. 
I'll have to check if a messenger exception etc. Is being explicitly cached at some point.

Reply:
I agree the manual code to update stack trace is best removed. I suspect it was a workaround for 1.4, or perhaps something that seemed correct and worked in one case but really in general does not work well. A custom exception type instead of Runtime, to give a better name to the exception, would be good. Something like "FatalGameException" or what not.